docs/sts/tls.md

```diff
@@ -0,0 +1,126 @@
+# AssumeRoleWithCertificate [![Slack](https://slack.min.io/slack?type=svg)](https://slack.min.io)
+
+## Introduction
+
+MinIO provides a custom STS API that allows authentication with client X.509 / TLS certificates.
+
+A major advantage of certificate-based authentication compared to other STS authentication methods,
+like OpenID Connect or LDAP/AD, is that client authentication works without any additional/external
+component that must be constantly available. Therefore, certificate-based authentication may provide
+better availability / lower operational complexity.
+
+The MinIO TLS STS API can be configured via MinIO's standard configuration API (i.e. using `mc admin config set/get`).
+Further, it can be configured via the following environment variables:
+
+```
+mc admin config set myminio identity_tls --env
+KEY:
+identity_tls  enable X.509 TLS certificate SSO support
+
+ARGS:
+MINIO_IDENTITY_TLS_STS_EXPIRY   (duration)  temporary credentials validity duration in s,m,h,d. Default is "1h"
```

Comment:
To keep it consistent with other STS APIs, we should use a "DurationSeconds" query parameter in the API to set per-call expiry, and remove this environment configuration. We are also using an upper limit of 1 year validity for all STS creds.

Reply:
It is used in the handler for AssumeRoleWithLDAP and in AssumeRoleWithWebIdentity - to set the expiry. Even AWS STS APIs have this parameter. Clients can set the expiry - AWS has a more restrictive upper limit, but we seem to have set it to 1 year now. @harshavardhana

Reply:
yeah AWS has an upper limit of 7 days we moved it to 365 days for a customer.

Reply:
> Hm... I don't see such a query parameter for e.g. `AssumeRoleWithLDAPIdentity`. Also, that would mean clients can set the expiry (within a time window themself). That does not seem correct...

We do we don't need a separate ENV to set the expiry - clients can ask for validity, we can of course reject STS requests that go beyond certificate validity if DurationSeconds lets say are beyond certificate not_after..

Reply:
That model seems odd. So a client can request credentials that are valid for an amount of time that is defined by the client itself. So a client can get credentials for 5 days or 10 days. I would assume the server defines the lifetime of the credentials, not the client.

I based `MINIO_IDENTITY_TLS_STS_EXPIRY` on https://github.com/minio/minio/blob/master/internal/config/identity/ldap/config.go#L99

Reply:
That ENV is going to be removed.